lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -901,6 +901,11 @@ NativeCodeGenerator::CodeGen(PageAllocator * pageAllocator, CodeGenWorkItem* wor
             workItem->GetJITData(),
             scriptContext->GetRemoteScriptAddr(),
             &jitWriteData);
+        if (hr == E_ACCESSDENIED && body->GetScriptContext()->IsClosed())
```

Comment:
agree we might need a custom hresult for it. currently WriteProcessMemory can also return E_ACCESSDENIED, however with scriptContext->IsClosed() check we can differentiate it and fail fast. but a custom hresult may be more clear